server/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -49,8 +51,8 @@
     public static final float DEFAULT_BOOST = 1.0f;
     public static final ParseField NAME_FIELD = new ParseField("_name");
     public static final ParseField BOOST_FIELD = new ParseField("boost");
-
-    private static int maxNestedDepth = 20;
+    //We set the default value for tests that don't go through SearchModule
+    private static int maxNestedDepth = SearchModule.INDICES_MAX_CLAUSE_COUNT_SETTING.getDefault(Settings.EMPTY);
```

Comment:
`INDICES_MAX_CLAUSE_COUNT_SETTING` is `4096`, which I don't think is correct.

I think you want `INDICES_MAX_NESTED_DEPTH_SETTING` here, which has a default value of `30` (changed in #90425)

Reply:
yes indeed, thanks for catching that. 4096 seems a bit high :) luckily this is only for tests. I tried to refactor it because I don't really like the static field and the current logic, but it's not so straight-forward to do.